src/Serializer/AbstractItemNormalizer.php

```diff
@@ -638,6 +639,17 @@ protected function getAttributeValue(object $object, string $attribute, string $
                 $childContext = $this->createChildContext($context, $attribute, $format);
                 unset($childContext['iri'], $childContext['uri_variables'], $childContext['resource_class'], $childContext['operation']);
 
+                if ($itemUriTemplate = $propertyMetadata->getUriTemplate()) {
```

Comment:
Would actually be cool if we can avoid populating `$attributeValue` before checking for uriTemplate. This can save quite some DB requests, as the actual data is not needed to generate the uri.

I made a proposal here: https://github.com/GregoireHebert/core/pull/1

This also includes enabling uriTemplate for non-collections. While we both have mainly collections in mind as the main purpose of this feature, I think there's no harm to also allow overriding the uriTemplate for *:1 relations.

Reply:
I am trying to take this into account, and it gives me insomnia and headaches...
I will eventually get there, please bear with me :D

Reply:
🤣 got it. Let me know if I can be of any help?

Reply:
I am getting somewhere... I think.
If you could check on your project that I dont break anything 😬
I had to change the HAL json schema that didn't allowed having object embedded (only array of objects)
But I wasn't 100% sure it was alright according to the specification.

Reply:
If it is ok, I'll write/update the documentation guide